File: trip_service/src/main/java/pl/edu/pg/trip/enity/Hotel.java
package pl.edu.pg.trip.enity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import pl.edu.pg.trip.listener.events.HotelDetailsResponse;

import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Hotel {
    private long id;
    private String place;
    private int stars;
    private String description;
    private String photo;
    private Set<HotelDetailsResponse.Room> rooms;
    private String airport;
    private String food;
    private String name;
}

File: tour_operator/hotels/Hotel.py
import json


class Hotel:
    def __init__(self, hotel):
        self.id = None
        hotel = json.loads(hotel)
        if "id" in hotel:
            self.id = hotel["id"]
        self.name = hotel["name"]
        self.stars = hotel["stars"]
        self.place = hotel["place"]
        self.photo = hotel["photo"]
        self.airport = hotel["airport"]

    def compare(self, hotel):
        """Compares Hotel object with HotelDetails object"""
        if self.name != hotel.name:
            return False
        if self.stars != hotel.stars:
            return False
        if self.photo != hotel.photo:
            return False
        if self.airport != hotel.airport:
            return False

        return True
